docs/guides/steth-integration-guide.md

```diff
@@ -175,19 +191,106 @@ When it hits the ground, transaction gets reverted.
+
+The amount of claimable ETH is determined once the withdrawal request is finalized. The rate stETH/ETH of the request finalization can't get higher than it's been at the moment of request creation. The user will be able to claim:
+
+- normally – ETH amount corresponding to the stETH amount at the moment of the request's placement
+
+**OR**
+
+- discounted - lowered ETH amount corresponding to the oracle-reported share rate in case the protocol had undergone significant losses (slashings and penalties)
+
+The second option is unlikely, and we haven't ever seen the conditions for it on mainnet so far.
+
+The end-user contract to deal with the withdrawals is `WithdrawalQueueERC721.sol`, which implements the ERC721 standard. NFT represents the position in the withdrawal queue and may be claimed after finalization of the request.
+
+Let's follow these steps in details:
+
+### Request and receive NFT
+
+You have several options for requesting for withdrawals, they requires you own stETH or wstETH on your address:
+
+- stETH
+    - Call `requestWithdrawalsWithPermit(uint256[] _amounts, address _owner, PermitInput _permit)` and get the ids of created positions, where `msg.sender` will be used to transfer tokens from and the `_owner` will be the address that can claim or transfer NFT (defaults to `msg.sender` if it’s not provided.
+    - Alternatively, sending stETH on behalf of `WithdrawalQueueERC721.sol` contract can be approved in a separate upfront transaction (`stETH.approve(withdrawalQueueERC712.address, allowance)`), and the `requestWithdrawals(uint256[] _amounts, address _owner)` method called afterwards.
+
+- wstETH
+    - Call `requestWithdrawalsWstETHWithPermit(uint256[] _amounts, address _owner, PermitInput _permit)` and get the ids of created positions, where `msg.sender` will be used to transfer tokens from, and the `_owner` will be the address that can claim or transfer NFT (defaults to `msg.sender` if it’s not provided.
+    - Alternatively, sending wstETH on behalf of `WithdrawalQueueERC721.sol` contract can be approved in a separate upfront transaction (`wstETH.approve(withdrawalQueueERC712.address, allowance)`), and the `requestWithdrawalsWstETH(uint256[] _amounts, address _owner)` method called afterwards.
+
+`PermitInput` structure defined as follows:
+
+```solidity
+struct PermitInput {
+    uint256 value;
+    uint256 deadline;
+    uint8 v;
+    bytes32 r;
+    bytes32 s;
+}
+```
+
+After request ERC721 NFT are minted to `_owner` address and can be transferred to the other owner which will have all the rights to claim the withdrawal.
+
+Additionally, this NFT implements [ERC4906] standard and it's recommended to rely on
+
+```solidity
+event BatchMetadataUpdate(uint256 _fromTokenId, uint256 _toTokenId);
+```
+
+to update the nft metadata if you're integrating it somewhere where it should be displayed correctly.
```

Comment:
maybe should explain a bit more (what are the states metadata can present and when updates)